### PR TITLE
fix schema + parsing for ads analytics metrics

### PR DESCRIPTION
**Pull request by AdamBkl**
## What
*Describe what the change is solving*
*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>

#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json [hunk 1]
```diff
@@ -1,109 +1,2638 @@
...
-      "type": "number"
+    "metrics": {
+      "type": "object",
+      "properties": {
+        "video_views_50": {
+          "type": "null"
+        },
+        "impressions": {
+          "type": "array",
+          "items": [
```
Comment (zfhui) [inline airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:24]: Oh... Do you have to map everything from an array into a type? That sounds very dirty. And how do you know how many items the array has?

Comment (AdamBkl) [reply airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:24]: As far as I understand from the docs we need the type for array items. You are totally right with this being dirty + I don't know the number of items the array has. I generated this schema from the response, I retrieved while running locally. I need to look further into this. 👀 

> Arrays contain 0 or more items, which must have a defined type. These types should also conform to the type system.` Arrays may require that all of their elements be the same type ("items": {whatever type...}), or they may require specific types for the first N entries ("items": [{first type...}, {second type...}, ... , {Nth type...}], AKA tuple-type).

https://docs.airbyte.com/understanding-airbyte/supported-data-types

Comment (dinks) [reply airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:24]: eh no.. I suggest you do 
```
"impressions": {
  "type": ["null", "array"],
  "items": {
    "type": ["null", "integer"]
  }
}
}
```

File: airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json [hunk 2]
```diff
@@ -1,109 +1,2638 @@
...
+            {
+              "type": "integer"
+            }
+          ]
+        },
+        "mobile_conversion_spent_credits": {
+          "type": "object",
+          "properties": {
+            "post_view": {
+              "type": "null"
```
Comment (dinks) [inline airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:166]: this seems to be wrong...

File: airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json (hunk 1 above)
Comment (AdamBkl) [reply airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json]: I fixed this as per dineshs suggestions

File: airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json [hunk 3]
```diff
@@ -1,109 +1,762 @@
...
-    "video_views_100": {
-      "type": "number"
-    },
-    "video_views_50": {
-      "type": "number"
-    },
-    "video_views_75": {
-      "type": "number"
+    "segment": {
+      "type": "object",
```
Comment (dinks) [inline airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:6]: `["null", "object"]` going along with our earlier conversation...

File: airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json [hunk 4]
```diff
@@ -1,109 +1,762 @@
...
+      "properties": {
+        "segment_name": {
+          "type": "string"
```
Comment (dinks) [inline airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:9]: Here too

File: airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json [hunk 5]
```diff
@@ -1,109 +1,762 @@
...
+        },
+        "segment_value": {
+          "type": "string"
```
Comment (dinks) [inline airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:12]: Here too

File: airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json [hunk 6]
```diff
@@ -1,109 +1,762 @@
...
+            "sale_amount": {
+              "type": ["null", "integer"]
+            }
+          }
+        },
+        "tweets_send": {
+          "type": "array",
+          "items": {
+            "type": ["null", "integer"]
+            }
```
Comment (dinks) [inline airbyte-integrations/connectors/source-twitter-ads/source_twitter_ads/schemas/ads_analytics_metrics.json:132]: Could you run an alignment on this file please?

**Review by dinks — APPROVED** [review]

**Review by dinks — APPROVED** [review]
Go

**Pull request merged**